File: src/main/java/org/example/healthcare/service/TranslationService.java
package org.example.healthcare.service;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;

import java.util.HashMap;
import java.util.Map;

@Service
public class TranslationService {
    
    private static final Logger logger = LoggerFactory.getLogger(TranslationService.class);
    
    @Value("${translation.api.enabled:false}")
    private boolean translationApiEnabled;
    
    @Value("${translation.api.url:}")
    private String translationApiUrl;
    
    @Value("${translation.api.key:}")
    private String translationApiKey;
    
    private final RestTemplate restTemplate = new RestTemplate();
    
    public String translate(String text, String sourceLang, String targetLang) {
        if (!translationApiEnabled || translationApiUrl.isEmpty()) {
            logger.warn("Translation API not enabled, returning original text");
            return text; // Fallback: return original text
        }
        
        try {
            Map<String, Object> request = new HashMap<>();
            request.put("text", text);
            request.put("source", sourceLang);
            request.put("target", targetLang);
            
            Map<String, String> headers = new HashMap<>();
            headers.put("Authorization", "Bearer " + translationApiKey);
            
            // This is a placeholder - actual implementation would call real translation API
            // For example: Google Translate API, DeepL API, etc.
            logger.info("Translating text from {} to {}", sourceLang, targetLang);
            
            // Mock response for now
            return text + " [Translated: " + sourceLang + " -> " + targetLang + "]";
        } catch (Exception e) {
            logger.error("Translation failed: {}", e.getMessage());
            return text; // Return original on error
        }
    }
    
    public String translateArabicToEnglish(String arabicText) {
        return translate(arabicText, "ar", "en");
    }
    
    public String translateEnglishToArabic(String englishText) {
        return translate(englishText, "en", "ar");
    }
}

File: frontend/src/services/api/translationService.ts
import apiClient from './apiClient'

export interface TranslationResponse {
  originalText: string
  translatedText: string
  sourceLang?: string
  targetLang?: string
}

export const translationService = {
  translate: async (text: string, sourceLang: string, targetLang: string): Promise<TranslationResponse> => {
    const response = await apiClient.post('/translate', {
      text,
      sourceLang,
      targetLang,
    })
    return response.data
  },

  translateArabicToEnglish: async (text: string): Promise<TranslationResponse> => {
    const response = await apiClient.post('/translate/ar-to-en', { text })
    return response.data
  },

  translateEnglishToArabic: async (text: string): Promise<TranslationResponse> => {
    const response = await apiClient.post('/translate/en-to-ar', { text })
    return response.data
  },
}

